### PR TITLE
reference/kubectl: Update docs for moving kubectl subresource to beta

### PR DESCRIPTION
**Pull request by MadhavJivrajani**
Ref: https://github.com/kubernetes/enhancements/issues/2590

/sig cli
/cc @nikhita

### CODE REVIEW

**Comment by mickeyboxell** [comment]
Hi @MadhavJivrajani  👋 please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review as soon as possible. 01:00 UTC Wednesday 22nd March 2023 / 17:00 PDT Tuesday 21st March 2023 is the official deadline.

This PR will need a doc review by Tuesday 4th April 2023 to get this into the release. Please reach out to required SIGs to get their review. Thank you!

**Comment by mickeyboxell** [comment]
@tengqm @sftim Would you mind taking a look at this when you have a moment?

**Comment by MadhavJivrajani** [comment]
Done @mickeyboxell , thank you!

File: content/en/docs/reference/kubectl/conventions.md [hunk 1]
```diff
@@ -22,9 +22,10 @@ For a stable output in a script:
 
 ## Subresources
 
-* You can use the `--subresource` alpha flag for kubectl commands like `get`, `patch`,
+* You can use the `--subresource` beta flag for kubectl commands like `get`, `patch`,
 `edit` and `replace` to fetch and update subresources for all resources that
 support them. Currently, only the `status` and `scale` subresources are supported.
+  * Please note that the `scale` subresource is not supported for use with the `--subresource` flag for the `kubectl edit` command.
```
Comment (sftim) [inline content/en/docs/reference/kubectl/conventions.md:28]: (nit)
Try something like this, where we explain what happens:
```markdown
  * For `kubectl edit`, the `scale` subresource is not supported. If you use  `--subresource` with
    `kubectl edit` and specify `scale` as a subresource, DEMONS FLY OUT OF YOUR NOSE.
```
(update **DEMONS FLY OUT OF YOUR NOSE** to match actual outcome)

**Comment by MadhavJivrajani** [comment]
@mickeyboxell @sftim PTAL, thank you!

**Pull request merged**